src/App.tsx

```diff
@@ -1,24 +1,203 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import {
+  FC,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  USER_ID as id,
+  getTodos,
+  postTodo,
+  deleteTodo,
+} from './api/todos';
+import Footer from './Components/Footer/Footer';
+import Header from './Components/Header/Header';
+import TodoList from './Components/TodoList/TodoList';
+import Notification from './Components/Notification/Notification';
+import Loader from './Components/Loader/Loader';
+import { Todo } from './types/Todo';
+import { Error, ErrorMessage } from './types/Error';
+import { FilterType } from './types/FilterType';
+import { filterTodoList, getMountCompletedTodos } from './utils/filterTodoList';
 
-const USER_ID = 0;
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [amountCompletedTodos, setAmountCompletedTodos] = useState(0);
+  const [error, setError] = useState<Error>({
+    status: false,
+    message: ErrorMessage.NONE,
+  });
+  const [isLoader, setIsLoader] = useState(true);
+  const [filterTodosBy, setFilterTodos] = useState(FilterType.All);
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [isDisabledAddingForm, setIsDisabledAddingForm] = useState(false);
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [deleteTodosId, setDeleteTodosId] = useState<number[] | null>(null);
+
+  const visibleTodoList = useMemo(() => (
+    filterTodoList(todos, filterTodosBy)
+  ), [todos, filterTodosBy]);
+
+  const fetchTodos = async () => {
+    try {
+      const todosFromServer = await getTodos(id);
+
+      setTodos(todosFromServer);
+      setAmountCompletedTodos(getMountCompletedTodos(todosFromServer));
+
+      if (!todosFromServer.length) {
+        setError({
+          status: true,
+          message: ErrorMessage.LOAD,
+        });
+      }
+    } catch {
+      setError({
+        status: true,
+        message: ErrorMessage.LOAD,
+      });
+    } finally {
+      setIsLoader(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+  }, []);
+
+  const addNewTodo = async (todo: string) => {
+    setIsDisabledAddingForm(true);
+    setError({
+      status: false,
+      message: ErrorMessage.NONE,
+    });
+    setIsAddingTodo(true);
+
+    setTempTodo({
+      id: Date.now(),
+      userId: id,
+      title: todo,
+      completed: false,
+    });
+
+    try {
+      const post = await postTodo(id, {
+        title: todo,
+        userId: id,
+        completed: false,
+      });
+      const resultTodos = [...todos, post];
+
+      setTodos(resultTodos);
```

Comment:
better to change todos based on prevTodos.
```suggestion
      setTodos(prevTodos => [...prevTodos, post]);
```